components/suggest/src/store.rs

```diff
@@ -601,6 +639,8 @@ struct SuggestStoreDbs {
     writer: SuggestDb,
     /// A read-only connection used to query the database.
     reader: SuggestDb,
+    /// A read-write connection used specifically for ingestion
+    ingestion: SuggestDb,
```

Comment:
I like the idea of adding a 3rd connection, but maybe we should put out a quick fix for the iOS issue and tackle this separately.  WDY2T?

Reply:
Oh, let's not have two read-write connections...having multiple writers is a bag of hurt in SQLite. We learned that the hard way with Places, where we saw contention between them—and, in Desktop, we could wedge one writer permanently, so that it always returned `SQLITE_BUSY` 😱

Reply:
Seems good for now, I updated the PR to do that.

Long-term, I still kind of like separating the connections since it allows you to cancel each separately.   The current `interrupt_ingestion` method could also interrupt a call to `dismiss_suggestion`.  I think there's a way to do it safely, although maybe it's just because I haven't been burned by SQLite yet.

Reply:
Thanks, @bendk, and I'm truly sorry if my reaction came off as short in any way—I was running a little late for an appointment earlier, and you deserved a more thoughtful response with more time. I left some more noodlings on https://bugzilla.mozilla.org/show_bug.cgi?id=1897299 for your consideration 🧠